src/components/chart/down-sample.ts

```diff
@@ -0,0 +1,67 @@
+import type { LineSeriesOption } from "echarts";
+
+export function downSampleLineData(
+  data: LineSeriesOption["data"],
+  chartWidth: number,
+  minX?: number,
+  maxX?: number
+) {
+  if (!data || data.length < 10) {
+    return data;
+  }
+  const width = chartWidth * window.devicePixelRatio;
+  if (data.length <= width) {
```

Comment:
Since we add 2 points per step, a min and a max, shouldnt we check if `data.length` is bigger than twice the width?

Wont we have cases where we end up with more points than the original data?